packages/@aws-cdk/cli-plugin-contract/lib/index.ts

```diff
@@ -0,0 +1,112 @@
+/**
+ * The basic contract for plug-ins to adhere to::
+ *
+ *   import { CustomCredentialProviderSource, CredentialProviderSourceRepository, Plugin } from '@aws-cdk/cli-plugin-contract';
+ *
+ *   export default class FooCDKPlugIn implements Plugin {
+ *     public readonly version = '1';
+ *
+ *     public init(host: CredentialProviderSourceRepository) {
+ *       host.registerCredentialProviderSource(new CustomCredentialProviderSource());
+ *     }
+ *   }
+ *
+ */
+export interface Plugin {
+  /**
+   * The version of the plug-in interface used by the plug-in. This will be used by
+   * the plug-in host to handle version changes.
+   */
+  version: '1';
+
+  /**
+   * When defined, this function is invoked right after the plug-in has been loaded,
+   * so that the plug-in is able to initialize itself. It may call methods of the
+   * `CredentialProviderSourceRepository` instance it receives to register new
+   * `CredentialProviderSource` instances.
+   */
+  init?: (host: CredentialProviderSourceRepository) => void;
+}
+
+export interface AwsCredentials {
+  /**
+   * AWS access key ID
+   */
+  readonly accessKeyId: string;
+
+  /**
+   * AWS secret access key
+   */
+  readonly secretAccessKey: string;
+
+  /**
+   * A security or session token to use with these credentials. Usually
+   * present for temporary credentials.
+   */
+  readonly sessionToken?: string;
+
+  /**
+   * AWS credential scope for this set of credentials.
+   */
+  readonly credentialScope?: string;
```

Comment:
Is this actually used?

Reply:
But we are the only consumers of this interface. We are describing V2 credentials, which are being plugged into a V3 SDK. So the credentials will never be plugged directly, we adapt them.

So we know exactly the part of the API that we expect.